src/v/kafka/server/handlers/configs/config_response_utils.cc

```diff
@@ -1031,4 +1031,27 @@ void report_broker_config(
       &describe_as_string<bool>);
 }
 
+std::vector<creatable_topic_configs> make_configs(
+  const cluster::metadata_cache& metadata_cache,
+  const cluster::topic_properties& topic_config) {
+    describe_configs_resource resource{};
+    describe_configs_result describe_result{};
+
+    report_topic_config(
+      resource, describe_result, metadata_cache, topic_config, false, false);
+
+    std::vector<creatable_topic_configs> result;
+    result.reserve(describe_result.configs.size());
+
+    for (auto& describe_conf : describe_result.configs) {
+        result.push_back(creatable_topic_configs{
+          .name = std::move(describe_conf.name),
+          .value = std::move(describe_conf.value),
+          .config_source = describe_conf.config_source,
+        });
+    }
+
+    return result;
+}
```

Comment:
Should this be a fragmented vector cc @BenPope ?  Worrying about the 1k topics create overallocation issue

Reply:
If that code was just copy-pasta'ed I think we should just keep it as is for now, since 95% of this PR is mainly moving around code so that it can invoked in two distinct places. 

If we decide to change that it should be in a follow up PR, IMO

Reply:
I'm surprised it didn't conflict with a change I made in https://github.com/redpanda-data/redpanda/pull/16982, but it looks like I missed this. Yeah, I think it should be `chunked_vector`.

Reply:
I've created an issue to take this up as a follow-up: https://github.com/redpanda-data/core-internal/issues/1174